mautic/mautic
# Fixed monitored inbox issue with selected folders and expanded feature

Please answer the following questions. 

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | y
| Related user documentation PR URL | na
| Related developer documentation PR URL | todo
| Issues addressed (#s or URLs) | #1472 
| BC breaks? | n
| Deprecations? | n

**Note that all new features should have a related user and/or developer documentation PR in their respective repositories.** 

### Required
#### Description:

1. This PR fixes a bug that caused the mautic:email:fetch command to not honor selected folders for monitored inboxes
2. Ok - I snuck in a feature. But it's cool and looking forward to leveraging it in future features of Mautic. But basically it extends the mautic:email:fetch command's support to allow a plugin/bundle to inject specific criteria for the messages desired whereas before, it only fetched unread emails. By default, only unread is fetched. 

#### Steps to test this PR:
1. Run the included unit tests
2. Code review the new feature as it's not leveraged yet - but the existing mautic:email:fetch command should still work
3. Manually test the fix using the steps below - only this time the unread message in the selected folder should be processed.

### As applicable
#### Steps to reproduce the bug:
1. Setup monitoring inbox in Configuration
2. Choose a folder other than INBOX for bounces
3. Ensure there is 1 unread message in the folder and no unread in inbox (or note the number of unread in the inbox - warning, this will set all your messages to read)
4. Run the command and note 0 emails processed (or the number you had in INBOX)
